File: RecipeRandomizer.Web/ClientApp/src/app/models/identity/registerRequest.ts
export class RegisterRequest {
    username: string;
    email: string;
    password: string;
    confirmPassword: string;
    hasAcceptedTerms: boolean;

    constructor(username: string, email: string, password: string, confirmPassword: string, hasAcceptedTerms: boolean) {
        this.username = username;
        this.email = email;
        this.password = password;
        this.confirmPassword = confirmPassword;
        this.hasAcceptedTerms = hasAcceptedTerms;
    }
}

File: RecipeRandomizer.Business/Models/Identity/RegisterRequest.cs
using System.ComponentModel.DataAnnotations;

namespace RecipeRandomizer.Business.Models.Identity
{
    public class RegisterRequest
    {
        [Required]
        public string FirstName { get; set; }

        [Required]
        public string LastName { get; set; }

        [Required]
        [EmailAddress]
        public string Email { get; set; }

        [Required]
        [MinLength(6)]
        public string Password { get; set; }

        [Required]
        [Compare("Password")]
        public string ConfirmPassword { get; set; }

        [Range(typeof(bool), "true", "true")]
        public bool HasAcceptedTerms { get; set; }
    }
}
